corona-warn-app/cwa-app-android
# FIX: Wong Output, when accessing OS push notification after pending test results turned positive (EXPOSUREAPP-11776)

This PR fixes `SubmissionTask`. When the user has both PCR and RAT tests waiting for submission and the user performs manual submission on one of these tests, `SubmissionTask` picks only the first test regardless of user selection. The second issue is when the test is in state (`isSubmitted == true`) we display the test result to the user in the `HomeFragment` even when we should display the test in `TestResultReady` state 

### TESTING
- scan pending PCR  and RAT tests without OS-Consent and without warning others
- change tests to positive 
- open later scanned test and perform the submission
- everything should work as expected
- if you have time check if notifications work as expected too, follow the steps in the bug ticket

### Jira Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11776
